File: 78-subsets/78-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        
        output = []
        
        def backtrack(index = 0, subset = []):
            output.append(subset.copy())
            for i in range(index, len(nums)):
                subset.append(nums[i])
                backtrack(i + 1, subset)
                subset.pop()
        
        backtrack()
        return output

File: 78-subsets/78-subsets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
    res = []
    helper(0, nums, [], res)
    return res
};

function helper(i, nums, path, res) {
    if (i == nums.length) {
        res.push(path.slice())
        return
    }
    helper(i + 1, nums, path, res)
    path.push(nums[i])
    helper(i + 1, nums, path, res)
    path.pop()
}